apis/v1alpha1/gateway_types.go

```diff
@@ -483,47 +485,125 @@ type GatewayStatus struct {
 	//
 	// These addresses should all be of type "IPAddress".
 	//
-	// +required
+	// +optional
```

Comment:
Instead of `Addresses` being optional, is their value in keeping it required and adding an "Unknown" `AddressType`?

Reply:
I don't think that's necessary, but I'd be happy to argue my case in a separate issue :D